language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/createvar/CreateVariableCodeAction.java

```diff
@@ -120,16 +128,17 @@ protected CreateVariableOut getCreateVariableTextEdits(Range range, DiagBasedPos
 
         String name = NameUtil.generateVariableName(matchedSymbol, typeDescriptor, allNameEntries);
 
-        List<TextEdit> importEdits = new ArrayList<>();
         List<TextEdit> edits = new ArrayList<>();
-        List<String> types = CodeActionUtil.getPossibleTypes(typeDescriptor, importEdits, context);
+        List<Integer> renamePositions = new ArrayList<>();
+        List<String> types = CodeActionUtil.getPossibleTypes(typeDescriptor, context, importsAcceptor);
         Position pos = range.getStart();
         for (String type : types) {
             Position insertPos = new Position(pos.getLine(), pos.getCharacter());
             String edit = type + " " + name + " = ";
             edits.add(new TextEdit(new Range(insertPos, insertPos), edit));
+            renamePositions.add(type.length() + 1);
         }
-        return new CreateVariableOut(name, types, edits, importEdits);
+        return new CreateVariableOut(name, types, edits, importsAcceptor.getNewImportTextEdits(), renamePositions);
```

Comment:
Can there be a chance where the same import edits being applied twice?

Reply:
Yes, in `Create variable and check error` code action, when we used `getTypeWithoutError()` in two different places to get the variable type without `error` the same import was applied twice. To avoid that, the importsAcceptor was used. Since we decided to calculate the length of the variable type inside `getCreateVariableTextEdits` itself we can remove these changes, but in my opinion this could be useful in the future. WDYT?